### ISSUE TITLE
operator new in declarations

### ISSUE THREAD

**Issue by kiselgra** [kiselgra/c-mera#33]
I was expecting that `new` in `decl` would work, however it gets parsed quite differently.

```
$ cat /tmp/cm.lisp 
(decl ((int *x (new 'int[10])))
  (set x (new 'int[10]))
  (set x (new (aref 'int 10))))
```

This is the output:

```
$ cm c++ /tmp/cm.lisp 
int *x { new int_10_ };
x = new int_10_;
x = new int[10];
```

@lispbub Do you think it would work to always parse the contents of new as decl-item?
Maybe we could also do something like with the names of decl-items to get the size of the array (if):
`(new unsigned long int 10)`, that would get us around the awkward `aref` syntax ;)

**Comment by kiselgra** [comment]
Hm, adding ctor arguments would also be required somehow, maybe having a list instead of a number for single instances with arguments?

**Comment by lispbub** [comment]
Fixed in c-mera 1.0

example:
```
(decl ((int *x = (new int[10])))
          (set x (new int[10]))
          (set x (new (aref int 10)))
          (set x (new unsgiend long (aref int 10)))
          (set x (new unsigned long int[10]))
          (set x (new (foo-bar a b c))))
```
yields
```
int *x = new int[10];
x = new int[10];
x = new int[10];
x = new unsgiend long int[10];
x = new unsigned long int[10];
x = new foo_bar(a, b, c);
```
